src/Policy/ResolverInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authorization\Policy;
+
+interface ResolverInterface
+{
+
+    /**
+     * Resolves the policy object based on the authorization resource.
+     *
+     * @param mixed $resource A resource that the access is checked against.
+     * @return object
+     */
+    public function getPolicy($resource);
```

Comment:
Should we have an `$action` as a second parameter? Somebody may want to map closures or invokable objects as policies for individual actions.

This would require having a policy handler. Default handler would call `canAction()` method on a policy, but somebody might implement a hadler that would invoke the policy object or call other method. The similar design is implemented by tactician: https://tactician.thephpleague.com/tweaking-tactician/

Reply:
> Should we have an $action as a second parameter? Somebody may want to map closures or invokable objects as policies for individual actions.

I don't think that is necessary. We've been planning on mapping policies to resources. Including actions muddies the purpose of the action parameter when running a policy.